### PR TITLE
Add typed side-data's

### PR DESCRIPTION
**Pull request by ZloyDyadka**
Added support for typed side-data, as they have a structure.
(The API has been changed, but it doesn't seem to be a problem since the latest implementation was not tagged)

### CODE REVIEW

**Comment by vansante** [comment]
Looks like a very high quality MR, I can't pick any faults 👍. There is however a linter issue, can you address that?

File: sidedata.go [hunk 1]
```diff
@@ -72,51 +81,56 @@ type SideDataMasteringDisplayMetadata struct {
...
+	SideDataBase
+	Data interface{} `json:"-"`
 }
 
-// Custom UnmarshalJSON to decode into specific structure based on SideDataType
 func (sd *SideData) UnmarshalJSON(b []byte) error {
-	var raw = &struct {
-		SideDataType string `json:"side_data_type"`
-	}{}
+	type Alias SideData
```
Comment (vansante) [inline sidedata.go:99]: Why is this type alias necessary? 🤔

Comment (ZloyDyadka) [reply sidedata.go:99]: This is necessary to avoid recursion, as the source type has a Marshal method.

Comment (ZloyDyadka) [reply sidedata.go:99]: > Why is this type alias necessary? 🤔

This is necessary to avoid recursion, as the source type has a Marshal method.

**Review by vansante — APPROVED** [review]
LGTM

**Pull request merged**